site/content/guides/setup-scripts.md

```diff
@@ -0,0 +1,145 @@
+
+> Like for [Browser checks](/docs/browser-checks), which libraries and modules are available in setup and teardown scripts depends on which [Runtime](/docs/runtimes) you have chosen. You can find a list of exactly [which libraries are included](/docs/runtimes/specs) in which runtime.
+
+### Prepare test data
+
+A check that is self-contained is responsible for preparing (and cleaning up - more on that in [teardown scripts](/docs/api-checks/setup-teardown-scripts/#teardown-scripts)) all the test data it needs to properly test the target functionality. Let's take a look at an example.
+
+We [use Checkly to monitor Checkly](https://blog.checklyhq.com/how-we-monitor-checkly/)! That includes [our API](/docs/api), which exposes endpoints for CRUD operations on checks, groups, alert channels and other resources. One of the checks we run verifies that our `DELETE /v1/checks/{id}` works correctly. In order to verify that, we will need a check to actually delete. To keep things nice and clean, we can create a new check in the setup script, then retrieve its unique `id` and pass it to the main HTTP request so we can tell our API exactly which check to delete.
+
+First off, let's look at the basic HTTP request config inside our new API check. Note that this will be the request to delete the existing check, not the one to create the dummy check that will be deleted: that will be featured in our setup script, which will run before the request shown below.
+
+Our main request will hit the `DELETE` endpoint at `https://api.checklyhq.com/v1/checks`:
+
+{{< figure src="/guides/images/guides-checkly-setup-delete.png" alt="checkly API check http request config" title="The basic config for our API check's HTTP request" >}}
+
+Note that we are including the `Authorization` header set to `Bearer <YOUR_CHECKLY_API_KEY>`. We will not need to set the body or other parameters.
+
+Let's now look at the setup script. The first thing we need to do is decide what library we will use for creating our dummy check, then import it. I will use {{< newtabref  href="https://axios-http.com/" title="axios" >}}:
+
+```js
+const axios = require("axios"); // import axios library explicitly
+```
+
+The basic axios config will set up a POST request to our [`CREATE CHECK` endpoint](https://www.checklyhq.com/docs/api#operation/postV1Checks) at `https://api.checklyhq.com/v1/checks`. We will set the `Authorization` header to pass our account's API key to authenticate ourselves.
+
+```js
+const { data } = await axios({ // create dummy check on Checkly
+  method: "post",
+  url: "https://api.checklyhq.com/v1/checks",
+  headers: {
+    Authorization: "Bearer YOUR_CHECKLY_API_KEY"
+  },
+  ...
+}
+```
+
+Note that you can use [environment variables](https://www.checklyhq.com/docs/api-checks/variables/) not to expose your credentials in the setup script:
```

Comment:
Next, we specify the body of our axios request, which states how the check we want to be created for us will look like: